x-pack/plugins/observability_ai_assistant/public/components/chat/chat_body.tsx

```diff
@@ -253,8 +254,9 @@ export function ChatBody({
         <EuiFlexItem grow={false}>
           <EuiPanel hasBorder={false} hasShadow={false} paddingSize="m">
             <ChatPromptEditor
-              loading={isLoading}
+              chatService={chatService}
```

Comment:
can we use a callback here rather than injecting the chatService? passing the chatService down makes it hard to figure out what it is used for and it makes testing harder

Reply:
Style wise do you prefer prop drilling with callbacks over using a context?

Reply:
Fair enough

Reply:
@miltonhultgren They're both not great, but if I had to choose I'd choose prop drilling over requiring context. Reason being it is clear from the one component's API what it needs to function. When your component requires a context you can't easily tell from its API that that's the case. Prop drilling may be slightly annoying, but at least it's explicit.

Reply:
💯